docs: add guide for NG0302

twerske opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?
Add Pipe Not Found guide and video (and link from framework thrown error)! (see: https://pr40580-35ec39c.ngbuilds.io/errors/NG0302)

![Screen Shot 2021-01-26 at 2 00 47 PM](https://user-images.githubusercontent.com/15061394/105911323-edaf5000-5fde-11eb-9df3-19a80519cde4.png)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

mary-poppins commented:
You can preview dabaaf1 at https://pr40580-dabaaf1.ngbuilds.io/.

AndrewKushnir reviewed (approved):
LGTM 👍

jessicajaniuk reviewed (approved):
Reviewed-for: public-api

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

petebacondarwin commented on aio/content/errors/NG0302.md:
```diff
@@ -0,0 +1,17 @@
+@shortDescription Pipe not found!
+
+@description
+Angular can’t find a pipe with this name. The pipe referenced in the template has not been named or declared properly. [Pipes](guide/pipes) must be either declared or imported in the `NgModule` where it is used, and the name used in a template must match the name defined in the pipe decorator. Pipes must be named uniquely in the `NgModule`.
```
> Pipes must be named uniquely within an `NgModule`.

??

petebacondarwin commented on aio/content/errors/NG0302.md:
```diff
@@ -0,0 +1,17 @@
+Use the pipe name to trace the templates or modules where this pipe is declared and used.
+
+To resolve this error, ensure that all pipes are properly:
+- Imported in the `NgModule` OR
+- Uniquely named and declared in the module file, if using a custom pipe
```
These two bullets seem a bit vague. Strictly we want something like:

* A local custom pipe is uniquely named and declared in the `NgModule`, or
* A pipe from another NgModule has that `NgModule` added to the `imports` of the `NgModule`.

petebacondarwin reviewed (approved):
A couple of nits but otherwise looks good.

Reviewed-for: public-api

mary-poppins commented:
You can preview f2daff4 at https://pr40580-f2daff4.ngbuilds.io/.

jelbourn reviewed (approved):
LGTM

Reviewed-for: public-api

gkalpak commented on aio/content/errors/NG0302.md:
```diff
@@ -0,0 +1,16 @@
+@name Pipe Not Found
+@category runtime
+@videoUrl https://www.youtube.com/embed/maI2u6Sxk9M
+@shortDescription Pipe not found!
```
OOC, why do some errors have exclamation points in the short descriptions?

twerske replied:
Exclamation points are in the errors where the message is a short sentence with an exclamation point to match for searching/discoverability (ex: "The pipe could not be found!") versus longer phased error messages (eg ExpressionChangedAfterChecked).

gkalpak commented on aio/content/errors/NG0302.md:
```diff
@@ -0,0 +1,16 @@
+@shortDescription Pipe not found!
+
+@description
+Angular can’t find a pipe with this name. The pipe referenced in the template has not been named or declared properly. [Pipes](guide/pipes) must be either declared or imported in the `NgModule` where it is used, and the name used in a template must match the name defined in the pipe decorator.
```
A couple of nits:
- Typo: `Pipes [...] where it is used` --> `A pipe [...] where it is used` (or `Pipes [...] where they are used`)?
- It is better to avoid "fancy" quotes (e.g. `’`), because they may render differently or incorrectly on some browsers.

```suggestion
Angular can't find a pipe with this name. The pipe referenced in the template has not been named or declared properly. A [pipe](guide/pipes) must be either declared or imported in the `NgModule` where it is used, and the name used in a template must match the name defined in the pipe decorator.
```

(BTW, it is easier for diffing/reviewing to have one sentence per line. This does not affect how the final HTML is rendered.)

twerske replied:
Good catch - TIL to break up each sentence on it's own line!

jessicajaniuk commented:
[Presubmit](http://test/OCL:354111317:BASE:354111342:1611769603623:de117b30) This shouldn't affect G3, but running presubmits anyway.

twerske committed:
docs: add guide for NG0302

new guide for pipe not found common error
link from pipe error to new guide
add embedded debugging video

mary-poppins commented:
You can preview 35ec39c at https://pr40580-35ec39c.ngbuilds.io/.
